src/main/java/com/google/sps/servlets/DashboardServlet.java

```diff
@@ -0,0 +1,107 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.sps.servlets.Property;
+import java.lang.reflect.Field; 
+import java.util.HashMap;
+import com.google.sps.servlets.RequestUtils;
+
+@WebServlet("/dashboard")
+public class DashboardServlet extends HttpServlet {
+
+  private final long MILLISECONDS_IN_DAY = 86400000;
+  private final String DEFAULT_START_TIMESTAMP = "0";
+  private final String DEFAULT_END_TIMESTAMP = String.valueOf(System.currentTimeMillis());
+
+ /**
+  * This route will return a JSON with the percentages of each interaction
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String startTimestamp = RequestUtils.getParameter(request, Property.TIMESTAMP);
+    String endTimestamp;
+
+    if (startTimestamp.isEmpty()) {
+      startTimestamp = DEFAULT_START_TIMESTAMP;
+      endTimestamp = DEFAULT_END_TIMESTAMP;
+    } else {
+      endTimestamp = String.valueOf(Long.valueOf(startTimestamp) + MILLISECONDS_IN_DAY);
```

Comment:
is startTime is not empty meaning a user session has begun, why do you set endTime to the start + milisecs in an entire day? Shouldnt endTime be when the user finishes his/her session?

Reply:
for the filter by date query, when the user selects a date, the timestamp that gets sent is the first timestamp of that day. so i'm adding the number of milliseconds in a day to figure out the range of timestamps for that date